app/kudos/views.py

```diff
@@ -755,3 +757,59 @@ def receive_bulk(request, secret):
         'tweet_text': urllib.parse.quote_plus(tweet_text)
     }
     return TemplateResponse(request, 'transaction/receive_bulk.html', params)
+
+
+def newkudos(request):
+    context = {
+        'active': 'newkudos',
+        'msg': None,
+    }
+
+    if not request.user.is_authenticated:
+        login_redirect = redirect('/login/github?next=' + request.get_full_path())
+        return login_redirect
+
+    if request.POST:
+        required_fields = ['name', 'description', 'priceFinney', 'artist', 'platform', 'numClonesAllowed', 'tags', 'to_address']
+        validtion_passed = True
+        for key in required_fields:
+            if not request.POST.get(key):
+                context['msg'] = str(_('You must provide the following fields: ')) + key
+                validtion_passed = False
+        if validtion_passed:
+            #upload to s3
+            img = request.FILES.get('photo')
+            session = boto3.Session(
+                aws_access_key_id=settings.AWS_ACCESS_KEY_ID,
+                aws_secret_access_key=settings.AWS_SECRET_ACCESS_KEY,
+            )
+
+            s3 = session.resource('s3')
+            key = f'media/uploads/{random.randint(999999999,99999999999)}_{img.name}'
```

Comment:
What do you think about using `uuid.uuid4` from the standard library? https://docs.python.org/3/library/uuid.html

Reply:
great idea

Reply:
updating